src/main/java/application/onlinebookstore/service/impl/BookServiceImpl.java

```
@@ -1,28 +1,41 @@
 package application.onlinebookstore.service.impl;
 
+import application.onlinebookstore.dto.BookDto;
+import application.onlinebookstore.dto.CreateBookRequestDto;
+import application.onlinebookstore.mapper.BookMapper;
 import application.onlinebookstore.model.Book;
 import application.onlinebookstore.repository.BookRepository;
 import application.onlinebookstore.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book savedBook = bookRepository.save(bookMapper.toModel(bookDto));
+        return bookMapper.toDto(savedBook);
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id));

```

Here you should throw an exception if there is no book for the provided ID